src/main/java/org/traccar/protocol/WialonProtocolDecoder.java

```diff
@@ -135,10 +135,41 @@ private Position decodePosition(Channel channel, SocketAddress remoteAddress, St
             for (String param : values) {
                 Matcher paramParser = Pattern.compile("(.*):[1-3]:(.*)").matcher(param);
                 if (paramParser.matches()) {
+                    // Parsing gives a (string,string) key-value pair
+                    String key = paramParser.group(1).toLowerCase();
+                    String value = paramParser.group(2);
+
+                    // Key is already in correct type (string)
+
+                    // If we can parse the value as a double, then we use that as the value's type.
+                    // This covers both integer (x:1:y) and double (x:2:y) types in the Wialon protocol
+
+                    // If not, value type is a string unless it is equal to some specific cases
+                    // (true, yes, etc), in which case we convert into a boolean
+
                     try {
-                        position.set(paramParser.group(1).toLowerCase(), Double.parseDouble(paramParser.group(2)));
+                        double double_value = Double.parseDouble(value);
+
+                        // Since accuracy is not part of the general parameter list,
+                        // we need to handle it separately by calling setAccuracy directly
+                        if (key.equals("accuracy")) {
+                            position.setAccuracy(double_value);
+                        }
+                        else {
+                            position.set(key, double_value);
+                        }
                     } catch (NumberFormatException e) {
-                        position.set(paramParser.group(1).toLowerCase(), paramParser.group(2));
+                        if (value.equalsIgnoreCase("true") || value.equalsIgnoreCase("t")
+                            || value.equalsIgnoreCase("yes") || value.equalsIgnoreCase("y")
+                            || value.equalsIgnoreCase("on")) {
```

Comment:
Why do we need to check for all these values?

Reply:
We don't. It's up to you to decide which values you'd like to check for.

Reply:
Maybe let's start with true/false for now.